Client/Frontend/Settings/SettingsTableViewController.swift

```diff
@@ -269,10 +289,18 @@ class StringSetting: Setting, UITextFieldDelegate {
             make.trailing.equalTo(cell.contentView).offset(-Padding)
             make.leading.equalTo(cell.contentView).offset(Padding)
         }
-        textField.text = prefs.stringForKey(prefKey) ?? defaultValue
+        textField.text = readPersistedValue() ?? defaultValue
         textFieldDidChange(textField)
     }
 
+    func readPersistedValue() -> String? {
+        fatalError("Override this method")
```

Comment:
This is a design smell…

Reply:
I introduced a SettingValuePersister protocol in https://github.com/mozilla-mobile/firefox-ios/pull/3491/files#diff-6209ecb7e871159425f66f16eed90d7dR254